File: ast-to-mermaid/tests/java/SwitchTest.java
// Test Java switch statement
public class SwitchTest {
    public static void main(String[] args) {
        int x = 2;

        switch (x) {
            case 1:
                System.out.println("One");
                break;
            case 2:
                System.out.println("Two");
                break;
            default:
                System.out.println("Other");
                break;
        }

        System.out.println("End of program");
    }
}

File: ast-to-mermaid/tests/cpp/switch_test.cpp
// Test C++ switch statement
#include <iostream>
using namespace std;

int main() {
    int x = 2;

    switch (x) {
        case 1:
            cout << "One";
            break;
        case 2:
            cout << "Two";
            break;
        default:
            cout << "Other";
    }

    cout << "End of program";
    return 0;
}